.github/workflows/generate_documentation.yml

```diff
@@ -0,0 +1,28 @@
+name: Generate documentation
+on:
+  push:
+    branches:
+    - master
+jobs:
+  generate_documentation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup cache
+        uses: actions/cache@v2
+        with:
+          path: $HOME/SpacemanDMM
+          key: ${{ runner.os }}-spacemandmm
+      - name: Install SpacemanDMM
+        run: bash tools/ci/install_spaceman_dmm.sh dmdoc
+      - name: Generate documentation
+        run: |
+          ~/dmdoc
+          touch dmdoc/.nojekyll
+          echo codedocs.tgstation13.org > dmdoc/CNAME
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          BRANCH: gh-pages
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          FOLDER: dmdoc
```

Comment:
consider CLEAN

Reply:
I saw this as an option but didn't quite understand the point, can you fill me in?

Reply:
You'd also use it with SINGLE_COMMIT likely. Basically, it ensures that the repository is only stuff generated by the script, and ensures files don't stick around that aren't needed anymore. I anticipated cache errors and stuff like that for Goon's, so I set it to both.

Reply:
Ah, I see. Good suggestion!